plugins/module_utils/nae.py

```diff
@@ -1108,6 +1117,9 @@ def new_compliance_requirement(self):
             self.result['Result'] = final_msg
 
     def new_compliance_requirement_set(self):
+        obj = self.query_compliance_object(json.loads(self.params.get('form')).get('name'))
```

Comment:
You might want to do a try/except around the json.loads in case it is not a proper JSON.
You might also want to verify that the return of json.loads is not None to be able to do a get.

Reply:
You might in fact want to put that in a function as you have called this 3 times already